cborg/src/Codec/CBOR/Read.hs

```diff
@@ -510,8 +510,8 @@ go_fast !bs da@(ConsumeNegWord64Canonical k) =
 go_fast !bs da@(ConsumeInt64Canonical k) =
   case tryConsumeInt64 (BS.unsafeHead bs) bs of
     DecodeFailure             -> go_fast_end bs da
-    DecodedToken sz i@(I64# i#)
-      | isInt64Canonical sz i  -> k i# >>= go_fast (BS.unsafeDrop sz bs)
+    DecodedToken sz (I64# i#)
+      | isInt64Canonical sz i# -> k i# >>= go_fast (BS.unsafeDrop sz bs)
```

Comment:
I'm confused by this. Did this simply never work?

Reply:
I would have to do some `git blame` archaeology to find out whether this worked in the past due to different type signatures or was just introduced without any testing; even more extremely, the code [didn't even parse on 32bit](https://github.com/well-typed/cborg/pull/322/files#diff-ff3003dee423ae4566843555c0bf4e38aea3df0d209ac5a8c53c58fb64b3bd16L1568-L1571) (introduced in #273) previously, so this code went without any testing since at least then.